scripts/event-relay.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const githubClient = require('../lib/github-client')
+
+async function handleJenkinsRelay (event) {
+  const { owner, repo, identifier } = event
+  const eventType = `jenkins.${identifier}.${event.event}`
+  try {
+    event.logger.debug(`Relaying ${eventType} to ${owner}/${repo}`)
+    await githubClient.repos.createDispatchEvent({
+      owner,
+      repo,
+      event_type: eventType,
+      client_payload: event
+    })
+    return true
+  } catch (err) {
+    event.logger.error(err, `Failed to relay ${eventType} to ${owner}/${repo}`)
+    return false
+  }
+}
+
+async function handleGitHubRelay (event, owner, repo) {
+  const { action } = event
+  const eventType = `github.${action}`
+  try {
+    event.logger.debug(`Relaying ${eventType} to ${owner}/${repo}`)
+    await githubClient.repos.createDispatchEvent({
+      owner,
+      repo,
+      event_type: eventType,
+      client_payload: event
+    })
+    return true
+  } catch (err) {
+    event.logger.error(err, `Failed to relay ${eventType} to ${owner}/${repo}`)
+    return false
+  }
+}
+
+module.exports = function (_, event) {
+  event.on('jenkins', handleJenkinsRelay)
+  event.on('github', handleGitHubRelay)
```

Comment:
Initially thought this would just end up being a GitHub event loopback (github.com -> nodejs-github-bot -> github.com), which doesn't seem to valuable from a glance. Have I misunderstood what's going on here?

Or are there use cases we can solve this way that cannot be done with GitHub Actions alone?

Reply:
The GitHub Relay works around GH limitations on Actions running on PRs from forks (meaning we'll be able to write Actions with write access and access to secrets for PRs).

Reply:
Aaaah thx!